nautobot_floor_plan/static/nautobot_floor_plan/js/floorplan.js

```diff
@@ -290,134 +304,426 @@ document.addEventListener('DOMContentLoaded', function() {
+    }
+
+    // Enable zoom and pan interactions
+    function enableZoomAndPan() {
+        // Restore mouse listeners for zoom and pan
+        document.querySelector("#floor-plan-svg").style.pointerEvents = "auto"; // Re-enable pointer events
+    }
+    // Function for complete cleanup of all highlighting effects
+    function completeCleanup(element, effectElements, animations) {
+        console.log("Performing complete cleanup of highlighting");
+    
+        // Stop all animations
+        animations.forEach(anim => {
+            if (anim && typeof anim.cancel === 'function') {
+                anim.cancel();
+            }
+        });
+    
+        // Remove all visual effect elements
+        effectElements.forEach(el => {
+            if (el && el.parentNode) {
+                el.parentNode.removeChild(el);
+            }
+        });
+    
+        // If element exists, restore its original styling
+        if (element) {
+            element.classList.remove('highlighted');
+            element.classList.remove('static-highlight');
+            element.removeAttribute('data-highlighted');
+        }
+    
+        console.log("Cleanup complete");
+    }
+
+    // Create all visual highlight effects and return references to animations and elements
+    function createHighlightEffects(element, svg) {
+        const effectElements = [];
+        const effectAnimations = [];
+
+        try {
+            // Get element's bounding box
+            const bbox = element.getBBox();
+            const centerX = bbox.x + bbox.width / 2;
+            const centerY = bbox.y + bbox.height / 2;
+
+            // Create a spotlight circle
+            const spotlight = document.createElementNS("http://www.w3.org/2000/svg", "circle");
+            spotlight.setAttribute("cx", centerX);
+            spotlight.setAttribute("cy", centerY);
+            spotlight.setAttribute("r", Math.max(bbox.width, bbox.height) * 1.5);
+            spotlight.setAttribute("fill", "rgba(255, 255, 0, 0.15)");
+            spotlight.setAttribute("class", "spotlight-effect");
+            spotlight.setAttribute("pointer-events", "none");
+            spotlight.setAttribute("data-highlight-effect", "true");
+
+            // Insert spotlight at the beginning of SVG
+            if (svg.firstChild) {
+                svg.insertBefore(spotlight, svg.firstChild);
+            } else {
+                svg.appendChild(spotlight);
+            }
+            effectElements.push(spotlight);
+
+            // Animate the spotlight - using opacity only to avoid r animation errors
+            try {
+                const spotlightAnim = spotlight.animate([
+                    { opacity: 0.1 },
+                    { opacity: 0.3 },
+                    { opacity: 0.1 }
+                ], {
+                    duration: 3000,
+                    iterations: Infinity
+                });
```

Comment:
```suggestion
                const spotlightAnim = spotlight.animate([
                    { opacity: 0.5 },
                    { opacity: 0.3 },
                    { opacity: 0.1 },
                    { opacity: 0.3 },
                    { opacity: 0.5 },
                    { opacity: 0.1 },
                ], {
                    delay: 300,
                    duration: 3000,
                    iterations: Infinity
                });
                effectAnimations.push(spotlightAnim);
```

Thoughts on this tweak to the spotlight?